client/hooks/useApiCall.js

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
```

Comment:
I don't understand the purpose of this hook. We should be using the trpc wrapper for react-query, which will handle all of these states.